api/groups/addressGroup.go

```diff
@@ -0,0 +1,707 @@
+	GetESDTsWithRole(address string, role string) ([]string, error)
+	GetAllESDTTokens(address string) (map[string]*esdt.ESDigitalToken, error)
+	GetKeyValuePairs(address string) (map[string]string, error)
+	IsInterfaceNil() bool
+}
+
+type addressGroup struct {
+	facade    addressFacadeHandler
+	mutFacade sync.RWMutex
+	*baseGroup
+}
+
+type esdtTokenData struct {
+	TokenIdentifier string `json:"tokenIdentifier"`
+	Balance         string `json:"balance"`
+	Properties      string `json:"properties"`
+}
+
+type esdtNFTTokenData struct {
+	TokenIdentifier string   `json:"tokenIdentifier"`
+	Balance         string   `json:"balance"`
+	Properties      string   `json:"properties,omitempty"`
+	Name            string   `json:"name,omitempty"`
+	Nonce           uint64   `json:"nonce,omitempty"`
+	Creator         string   `json:"creator,omitempty"`
+	Royalties       string   `json:"royalties,omitempty"`
+	Hash            []byte   `json:"hash,omitempty"`
+	URIs            [][]byte `json:"uris,omitempty"`
+	Attributes      []byte   `json:"attributes,omitempty"`
+}
+
+// NewAddressGroup returns a new instance of addressGroup
+func NewAddressGroup(facadeHandler interface{}) (*addressGroup, error) {
+	if facadeHandler == nil {
+		return nil, errors.ErrNilFacadeHandler
+	}
+
+	facade, ok := facadeHandler.(addressFacadeHandler)
+	if !ok {
+		return nil, fmt.Errorf("%w for address group", errors.ErrFacadeWrongTypeAssertion)
+	}
+
+	ag := &addressGroup{
+		facade:    facade,
+		baseGroup: &baseGroup{},
+	}
+
+	endpoints := []*shared.EndpointHandlerData{
+		{
+			Path:    getAccountPath,
+			Method:  http.MethodGet,
+			Handler: ag.getAccount,
+		},
+		{
+			Path:    getBalancePath,
+			Method:  http.MethodGet,
+			Handler: ag.getBalance,
+		},
+		{
+			Path:    getUsernamePath,
+			Method:  http.MethodGet,
+			Handler: ag.getUsername,
+		},
+		{
+			Path:    getKeyPath,
+			Method:  http.MethodGet,
+			Handler: ag.getValueForKey,
+		},
+		{
+			Path:    getKeysPath,
+			Method:  http.MethodGet,
+			Handler: ag.getKeyValuePairs,
+		},
+		{
+			Path:    getESDTBalancePath,
+			Method:  http.MethodGet,
+			Handler: ag.getESDTBalance,
+		},
+		{
+			Path:    getESDTNFTDataPath,
+			Method:  http.MethodGet,
+			Handler: ag.getESDTNFTData,
+		},
+		{
+			Path:    getESDTTokensPath,
+			Method:  http.MethodGet,
+			Handler: ag.getAllESDTData,
+		},
+		{
+			Path:    getRegisteredNFTsPath,
+			Method:  http.MethodGet,
+			Handler: ag.getNFTTokenIDsRegisteredByAddress,
+		},
+		{
+			Path:    getESDTTokensWithRolePath,
+			Method:  http.MethodGet,
+			Handler: ag.getESDTTokensWithRole,
+		},
+		{
+			Path:    getESDTsRolesPath,
+			Method:  http.MethodGet,
+			Handler: ag.getESDTsRoles,
+		},
+	}
+	ag.endpoints = endpoints
+
+	return ag, nil
+}
+
+// addressGroup returns a response containing information about the account correlated with provided address
+func (ag *addressGroup) getAccount(c *gin.Context) {
+	addr := c.Param("address")
```

Comment:
Lines `L150-L162` are just fetching the "address" parameter and respond with an error otherwise. Also, I cannot extract it as there are specific errors used (ErrGetBalance, ErrGetNonce, and so on)